internal/controlplane/handlers_oauth.go

```diff
@@ -288,6 +295,11 @@ func (s *Server) StoreProviderToken(ctx context.Context,
 		return nil, providerError(err)
 	}
 
+	if !slices.Contains(provider.AuthFlows, db.AuthorizationFlowUserInput) {
```

Comment:
why not call p.SupportsAuthFlow() here? is it because one is using a protobuf constant and the other a db constant? Since we're using the constant verbatim and not from a variable, could we use the protobuf constant here as well?

Reply:
I think `getProviderFromRequestOrDefault` actually returns a `db.Provider`, not a proto object mentioned in `SupportsAuthFlow`.

With that said, I'd prefer to have it return something (maybe a `ProviderBuilder`?) that encapsulates this check more than `slices.Contains` on a returned database row.

Reply:
Should we have a separate PR with that refactor?

Reply:
I'm fine with a separate PR if it comes soon and we don't propagate code patters like this over time

Reply:
on it.